pandas/tests/io/excel/test_writers.py

```diff
@@ -936,6 +936,17 @@ def test_to_excel_empty_multiindex(self, tmp_excel):
             result, expected, check_index_type=False, check_dtype=False
         )
 
+    def test_to_excel_empty_multiindex_both_axes(self, tmp_excel):
+        # GH 57696
+        df = DataFrame(
+            [],
+            index=MultiIndex.from_tuples([], names=[0, 1]),
+            columns=MultiIndex.from_tuples([("A", "B")]),
+        )
+        df.to_excel(tmp_excel)
+        result = pd.read_excel(tmp_excel, header=None)
+        assert result.iloc[:2, -1].to_list() == ["A", "B"]
```

Comment:
Can you test the entire result? I think if you change the read line to

    result = pd.read_excel("tmp.xlsx", header=[0, 1], index_col=[0, 1])

Then you should just be able to do

    tm.assert_frame_equal(result, df)

Reply:
thanks, that looks better and works exactly like you said.